pennylane_lightning/src/tests/Test_Util.cpp

```diff
@@ -386,3 +386,101 @@ TEMPLATE_TEST_CASE("Utility math functions", "[Util]", float, double) {
         }
     }
 }
+
+int popcount_slow(uint64_t x) {
+    int c = 0;
+    for (; x != 0; x >>= 1) {
+        if ((x & 1U) != 0U) {
+            c++;
+        }
+    }
+    return c;
+}
+
+int ctz_slow(uint64_t x) {
+    int c = 0;
+    while ((x & 1) == 0) {
+        x >>= 1;
+        ++c;
+    }
+    return c;
+}
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEST_CASE("Utility bit operations", "[Util]") {
+    SECTION("Internal::countBit1Fast") {
+        { // for uint32_t
+            uint32_t n = 0;
+            CHECK(Util::Internal::countBit1(n) == 0);
+            for (int k = 0; k < 100; k++) {
```

Comment:
Will this overflow the n value?

Reply:
https://en.cppreference.com/w/cpp/language/operator_arithmetic It seems like unsigned arithmetic will not overflow. I will make all operations unsigned tho as the [conversion rule](https://en.cppreference.com/w/cpp/language/implicit_conversion#Integral_conversions) is always confusing